File: intro_to_business_analytics/i2ba_chap6.R
library(dplyr)
as_tibble((mtcars))

data()

library(nycflights13)

flights = flights

write.csv(flights, 'flights.csv')

flights |> 
  filter(day == 1, month == 1)

flights |> 
  arrange(year, month, day) |> 
  arrange(desc(arr_delay))

flights |> 
  select(year, month, day)

flights |> 
  select(-(year:day))

flights |> 
  select(tailnum) |> 
  distinct()

flightSpeedDF <- flights |> 
  select(distance, air_time) |> 
  mutate(speed = distance / air_time * 60)

flights |> 
  summarize(
    delay = mean(dep_delay, na.rm = TRUE)
  )

destinations <- flights |> 
  group_by(dest)

destDF <- destinations |> 
  summarize(
    planes = n_distinct(tailnum),
    flights = n()
  )

lightSpeedDF4 <- flights |> 
  select(distance, air_time, dest) |> 
  mutate(speed = distance / air_time *60) |> 
  group_by(dest) |> 
  summarize(avgSpeed = mean(speed, na.rm = TRUE)) |> 
  arrange(desc(avgSpeed))

File: intro_to_business_analytics/i2ba_chap6.py
import polars as pl

flights = pl.read_csv('flights.csv', ignore_errors=True)
print(flights)

(
    flights
    .select("distance", "air_time", "dest")
    .with_columns(speed = pl.col("distance")/pl.col("air_time") * 60)
    .group_by(pl.col("dest")).agg(
        avgSpeed = pl.col("speed").mean()
    )
    .sort("avgSpeed", descending=True)
)
